test/dlc_tests/ec2/test_smdebug.py

```diff
@@ -10,73 +10,70 @@
-        pytest.skip("Currently skipping for TF2.3.0 on p2.8xlarge until the issue is fixed")
-    if is_tf1(training):
-        pytest.skip("Currently skipping for TF1 until the issue is fixed")
-    run_smdebug_test(training, ec2_connection, region, docker_executable="nvidia-docker", container_name="smdebug-gpu",
-                     is_large_shm="p2.8xlarge" in SMDEBUG_EC2_GPU_INSTANCE_TYPE)
+def test_smdebug_gpu(training, ec2_connection, region, ec2_instance_type, gpu_only, py3_only):
+    run_smdebug_test(
+        training,
+        ec2_connection,
+        region,
+        ec2_instance_type,
+        docker_executable="nvidia-docker",
+        container_name="smdebug-gpu",
+    )
 
 
 @pytest.mark.flaky(reruns=0)
 @pytest.mark.integration("smdebug")
 @pytest.mark.model("mnist")
 @pytest.mark.parametrize("ec2_instance_type", SMDEBUG_EC2_CPU_INSTANCE_TYPE, indirect=True)
-def test_smdebug_cpu(training, ec2_connection, region, cpu_only, py3_only):
-    # TODO: Remove this once test timeout has been debugged (failures especially on m4.16xlarge)
-    if is_tf2(training) and "m4.16xlarge" in SMDEBUG_EC2_CPU_INSTANCE_TYPE:
-        pytest.skip("Currently skipping for TF2 on m4.16xlarge until the issue is fixed")
-    if is_tf1(training):
-        pytest.skip("Currently skipping for TF1 until the issue is fixed")
-    run_smdebug_test(training, ec2_connection, region)
+def test_smdebug_cpu(training, ec2_connection, region, ec2_instance_type, cpu_only, py3_only):
+    run_smdebug_test(training, ec2_connection, region, ec2_instance_type)
+
+
+class SMDebugTestFailure(Exception):
+    pass
 
 
 def run_smdebug_test(
     image_uri,
     ec2_connection,
     region,
+    ec2_instance_type,
     docker_executable="docker",
     container_name="smdebug",
     test_script=SMDEBUG_SCRIPT,
-    logfile="output.log",
-    is_large_shm=False
 ):
+    large_shm_instance_types = ("p2.8xlarge", "m4.16xlarge")
+    shm_setting = " --shm-size=1g " if ec2_instance_type in large_shm_instance_types else " "
     framework = get_framework_from_image_uri(image_uri)
     container_test_local_dir = os.path.join("$HOME", "container_tests")
     ec2_connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
 
-    shm_setting = '--shm-size=1g' if is_large_shm else ""
-    ec2_connection.run(
-        f"{docker_executable} run {shm_setting} --name {container_name} -v "
-        f"{container_test_local_dir}:{os.path.join(os.sep, 'test')} -itd {image_uri}",
-        hide=True,
-    )
-
     try:
-        test_output = ec2_connection.run(
-            f"{docker_executable} exec --user root {container_name} "
-            f"/bin/bash -c '{test_script} {framework}' | tee {logfile}",
+        ec2_connection.run(
+            f"{docker_executable} run --name {container_name} -v "
+            f"{container_test_local_dir}:{os.path.join(os.sep, 'test')}{shm_setting}{image_uri} "
```

Comment:
nit: Why put the spaces in the `shm_setting` string, rather than placing the spaces in this main string here?

I think it is more readable if the arguments are properly spaced in this string.

Reply:
These are the scenarios I see:

scenario 1:
```
shm_setting = "--shm-size=1g" if ec2_instance_type in large_shm_instance_types else ""
...
f"{container_test_local_dir}:{os.path.join(os.sep, 'test')} {shm_setting} {image_uri} "
```
In this scenario, you have 2 empty spaces between two arguments - this may be harmless, but better to avoid

scenario 2: 
```
shm_setting = "--shm-size=1g " if ec2_instance_type in large_shm_instance_types else ""
...
f"{container_test_local_dir}:{os.path.join(os.sep, 'test')} {shm_setting}{image_uri} "
```
This could also go the other way (i.e. space can be before). This seems like the least readable option to me.

scenario 3: 
```
shm_setting = " --shm-size=1g " if ec2_instance_type in large_shm_instance_types else " "
...
f"{container_test_local_dir}:{os.path.join(os.sep, 'test')}{shm_setting}{image_uri} "
```
How it is currently implemented - avoids scenario adding 2 spaces, more consistent than scenario 2

Reply:
I think scenario 1 is acceptable. But I can live with scenario 3.

Reply:
Scenario 1 can introduce functional problems. Let's go with scenario 3.